.github/workflows/build.yml

```diff
@@ -333,3 +333,161 @@ jobs:
         with:
           name: code-coverage-report
           path: cobertura.xml
+
+  e2e-tests:
+    runs-on: ${{ matrix.config.os }}
+    needs: build-rust
+    ##  we can't run e2e tests consecutively due to 5 apps limit on cloud and the fact that we use same account for deploying and testing
+    concurrency:
+      group: e2e-tests
+    strategy:
+      matrix:
+        config:
+          - {
+              os: "ubuntu-22.04",
+              tinygo: "https://github.com/tinygo-org/tinygo/releases/download/v0.25.0/tinygo_0.25.0_amd64.deb",
+              clang: "https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-14/wasi-sdk-14.0-linux.tar.gz",
+              zig: "https://ziglang.org/download/0.10.0/zig-linux-x86_64-0.10.0.tar.xz",
+              grain: "https://github.com/grain-lang/grain/releases/download/grain-v0.5.4/grain-linux-x64",
+              zigpath: "zig-linux-x86_64-0.10.0"
+            }
+
+    services:
+      selenium:
+        image: selenium/standalone-chrome:4.1.2
+        options: --shm-size=2gb
+        ports:
+          - 4444:4444
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Retrieve saved Spin Binary
+        uses: actions/download-artifact@v3
+        with:
+          name: spin-ubuntu-latest
+          path: target/debug/
+
+      - name: Fix Spin Binary permissions
```

Comment:
Curious about this step; the binary appears to come with executable perms by default when I build locally.

Reply:
it was a copy paste from the existing workflow step :). I'll remove this step (will verify as well).

Reply:
so i checked, and it seems the executable permissions are not there, (https://github.com/actions/upload-artifact/issues/38).